homeassistant/data_entry_flow.py

```diff
@@ -684,6 +700,36 @@ def async_remove(self) -> None:
     async def async_setup_preview(hass: HomeAssistant) -> None:
         """Set up preview."""
 
+    @callback
+    def async_cancel_progress(self) -> None:
+        """Cancel in progress task."""
+        if self.__progress_task and not self.__progress_task.done():
+            self.__progress_task.cancel()
+        self.__progress_task = None
+        self.progress_fut = None
+
+    @callback
+    def async_start_progress(
+        self,
+        progress_job: Coroutine[Any, Any, None],
+        progress_done: Callable[..., Coroutine[Any, Any, Any]],
+    ) -> None:
+        """Start in progress task if not started."""
+
+        async def _send_event_when_done(fut: asyncio.Future[None]) -> None:
+            if not fut.done():
+                try:
+                    await fut
+                except Exception as err:  # pylint: disable=broad-exception-caught
+                    fut.set_exception(err)
```

Comment:
If we wrap this in a task instead we can simply await the job and let exceptions raise. Use try... finally to call `progress_done`.